server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -497,7 +499,8 @@ protected Node(
                     searchModule.getNamedXContents().stream(),
                     pluginsService.filterPlugins(Plugin.class).stream().flatMap(p -> p.getNamedXContent().stream()),
                     ClusterModule.getNamedXWriteables().stream(),
-                    SystemIndexMigrationExecutor.getNamedXContentParsers().stream()
+                    SystemIndexMigrationExecutor.getNamedXContentParsers().stream(),
+                    HealthNodeSelectorTaskExecutor.getNamedXContentParsers().stream()
```

Comment:
I think all changes in this class should be conditioned by the feature flag being enabled. 

What do you think?

Reply:
I was trying to avoid licking the flag outside of the `HealthNodeSelectorLifecycleHandler` but on second thought, I agree with putting everything behind the flag.